### PR TITLE
[73] Added features for industries in `CrunchbaseWrangler`

### LINKED ISSUE #73: Adding industries to `CrunchbaseWrangler`

### PR DESCRIPTION
**Pull request by beingkk**
Closes #73 

Adds functions in `CrunchbaseWrangler` class to find and filter companies related to Crunchbase's 700+ industries 

Usage is presented in the notebook `innovation_sweet_spots/analysis/examples/03_getting_cb_data.py`

---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

### CODE REVIEW

**Comment by beingkk** [comment]
@Jack-0-0 Hey Jack, these set of functions will be useful if you'd like to look at not only the "sectors" we've identified in the ISS pilot project on green tech, but also any other sector/industry specified by Crunchbase labels.

**Comment by beingkk** [comment]
Hey @Jack-0-0 , FYI I've changed the base to `dev` for this PR and resolved the conflicts.

File: innovation_sweet_spots/analysis/wrangling_utils.py [hunk 1]
```diff
@@ -134,9 +136,26 @@ class CrunchbaseWrangler:
...
+        self._cb_category_groups = None
+        self._cb_organisation_categories = None
+        # Organisation categories (industries)
+        self._industries = None
+        self._industry_groups = None
+        self._industry_to_group = None
+        self._group_to_industries = None
+
+    @property
+    @lru_cache(1)
```
Comment (Jack-0-0) [inline innovation_sweet_spots/analysis/wrangling_utils.py]: For `@lru_cache` why is it using `maxsize=1` ?

File: innovation_sweet_spots/analysis/wrangling_utils.py [hunk 2]
```diff
@@ -164,6 +183,95 @@ def cb_investors(self):
...
+                            *self.cb_category_groups["industry_groups"].to_list()
+                        )
+                    )
+                )
+            )
+        return self._industry_groups
+
+    @property
+    def industry_to_group(self):
+        if self._industry_to_group is None:
```
Comment (Jack-0-0) [inline innovation_sweet_spots/analysis/wrangling_utils.py]: There's a missing docstring here.

File: innovation_sweet_spots/analysis/wrangling_utils.py [hunk 3]
```diff
@@ -164,6 +183,95 @@ def cb_investors(self):
...
+                zip(
+                    self.cb_category_groups.industry,
+                    self.cb_category_groups.industry_groups.to_list(),
+                )
+            )
+        return self._industry_to_group
+
+    @property
+    def group_to_industries(self):
+        if self._group_to_industries is None:
```
Comment (Jack-0-0) [inline innovation_sweet_spots/analysis/wrangling_utils.py]: Also a missing docstring here.

File: innovation_sweet_spots/tests/test_wrangling_utils.py [hunk 4]
```diff
@@ -158,3 +158,19 @@ def test_convert_deal_currency_to_gbp():
...
     assert output_df.equals(expected_df)
+    
+
+### Testing other functions
+
+
+def test_split_comma_seperated_string():
+    assert split_comma_seperated_string("A, B,c") == ["A", "B", "c"]
+    assert split_comma_seperated_string("a,,b") == ["a", "", "b"]
+
```
Comment (Jack-0-0) [inline innovation_sweet_spots/tests/test_wrangling_utils.py]: Could add a case here for when `split_comma_seperated_string` returns `[]`

File: innovation_sweet_spots/analysis/wrangling_utils.py [hunk 5]
```diff
@@ -326,7 +434,113 @@ def get_organisation_investors(
...
+        )
+
 
 def get_years(dates: Iterator[datetime.date]) -> Iterator:
     """Converts a list of datetimes to years"""
     return [x.year for x in dates]
+
+
+def split_comma_seperated_string(text: str) -> Iterator[str]:
+    return [s.strip() for s in text.split(",")] if type(text) is str else []
```
Comment (Jack-0-0) [inline innovation_sweet_spots/analysis/wrangling_utils.py]: Missing a docstring here.

**Review by Jack-0-0 — COMMENTED** [review]
Thank you for adding this @beingkk. The notebook runs as expected. I've added some comments on specific lines.

File: innovation_sweet_spots/analysis/wrangling_utils.py (hunk 1 above)
Comment (beingkk) [inline innovation_sweet_spots/analysis/wrangling_utils.py:153]:
```suggestion
    @lru_cache
```

File: innovation_sweet_spots/analysis/wrangling_utils.py [hunk 6]
```diff
@@ -4,8 +4,10 @@
 from currency_converter import CurrencyConverter
 from datetime import datetime
 from typing import Iterator
+import itertools
 import innovation_sweet_spots.getters.gtr as gtr
 import innovation_sweet_spots.getters.crunchbase as cb
+from functools import lru_cache
```
Comment (beingkk) [inline innovation_sweet_spots/analysis/wrangling_utils.py]:
```suggestion
```

File: innovation_sweet_spots/analysis/wrangling_utils.py (hunk 1 above)
Comment (beingkk) [inline innovation_sweet_spots/analysis/wrangling_utils.py]:
```suggestion
```

Comment (beingkk) [reply innovation_sweet_spots/analysis/wrangling_utils.py]: Good question, I was experimenting a bit with lru_cache here, but I don't think I actually need it here. So, will delete  this line.

File: innovation_sweet_spots/tests/test_wrangling_utils.py (hunk 4 above)
Comment (beingkk) [reply innovation_sweet_spots/tests/test_wrangling_utils.py]: Nice suggestion, thanks!

**Comment by beingkk** [comment]
Thank you very much @Jack-0-0, I've fixed the issues you've raised and will merge this into `dev`.

**Pull request merged**